File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> freq;

        for (int n : nums) {
            freq[n]++;
        }

        vector<vector<int>> buckets(nums.size() + 1);

        for (auto& [num, count] : freq) {
            buckets[count].push_back(num);
        }

        vector<int> res;

        for (int count = nums.size(); count >= 1; count--) {
            for (int num : buckets[count]) {
                res.push_back(num);

                if (res.size() == k) {
                    return res;
                }
            }
        }

        return res;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        l = 1
        p = [1] * n
        for i in range(n):
            p[i] = l
            l = l * nums[i]
        
        r=1
        for i in range(n-1, -1, -1):
            p[i] = p[i] * r
            r = r * nums[i]
        return p